ext/search_kit/tests/phpunit/api/v4/SearchDisplay/SearchRunTest.php

```diff
@@ -1697,6 +1697,45 @@ public function testContributionAggregateCurrency():void {
     $this->assertEquals('$250.00', $result[3]['columns'][0]['val']);
   }
 
+  public function testContributionTotalCountWithTestAndTemplateContributions():void {
+    // Add a source here for the where below, as if we use id, we get the test and template contributions
+    $contributions = $this->saveTestRecords('Contribution', [
+      'records' => [
+        ['is_test' => TRUE, 'source' => 'TestTemplate'],
+        ['is_template' => TRUE, 'source' => 'TestTemplate'],
+        ['source' => 'TestTemplate'],
+      ],
+    ]);
+
+    $params = [
+      'checkPermissions' => FALSE,
+      'return' => 'tally',
+      'savedSearch' => [
+        'api_entity' => 'Contribution',
+        'api_params' => [
+          'version' => 4,
+          'select' => ['id'],
+          'where' => [['source', '=', 'TestTemplate']],
+        ],
+      ],
+      'display' => [
+        'settings' => [
+          'columns' => [
+            [
+              'key' => 'id',
+              'tally' => [
+                'fn' => 'COUNT',
+              ],
+            ],
+          ],
+        ],
+      ],
+    ];
+
+    $result = civicrm_api4('SearchDisplay', 'run', $params);
+    $this->assertEquals(1, $result[0]['id']);
```

Comment:
Test looks good.
Could you also run it with the same params but set `$params['return'] = 'page:1'` and check that it returns one row. That would feel like the test covers your bug report where the tally doesn't match the row count.

Reply:
Yes, done.

Reply:
Great thanks